### PR TITLE
Phil's batch of fixes

### LINKED ISSUE #12393: Lizards can shave their head with the electric razor.
The first time, it acts like there's a hair object there being shaved. The second time is when it says "there's no hair," which should happen ALL the time.

### LINKED ISSUE #12410: Slimes can continue to feed while jumping into a vent
Feed on something as a slime then vent crawl while feeding. You will still be eating while in the pipes though you cannot move since buckled

### LINKED ISSUE #12423: You cannot get seeds from Snapcorn
Server: Sybil and Basil

Revision: What revision was the server running? 2015-10-13
1d4cc5fea7a599aa19f5c450d359ef4503f968fc 

Problem Description: When trying to extract seeds from any part of the snap corn (the full plant itself, the empty cob, the resulting snaps) the seed extractor will tell you that it cannot extract seeds from it.

What did you expect to happen: I expected a full cob to produce seeds from a seed extractor

What happened instead: No seeds came forth.

Why is this bad/What are the consequences: Impossible to produce meaningful quantities of snap corn, inconsistent with all other plants.

Steps to reproduce the problem: Mutate corn into snapcorn, make sure the yield is >0, harvest and try and use resulting snap corn in the seed extractor.

Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): I'm pretty sure noone else has discovered this because noone else has ever made snapcorn.

### LINKED ISSUE #12437: Deconstructing chairs gives 2 metal sheets instead of 1.
Server: Sybil

Revision: 2015-10-16
cb58e8dca360b16f6fb362b29ab3e2d367f5db86 

Problem Description: When you build and deconstruct a chair you double the metal used to make the chair. Untested with any chair except the standard one.

What did you expect to happen: To deconstruct a chair and get 1 metal back.

What happened instead: I got 2 back

Why is this bad/What are the consequences: Infinite free metal from 1 sheet.

Steps to reproduce the problem: Build chair, wrench chair.

Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): Pretty ordinary round up until that point.

### LINKED ISSUE #12444: Portable chem dispenser is broken
It has 'undefined units' of energy and no buttons to dispense chemicals.

### LINKED ISSUE #12448: Botany winter coat hood is also a bucket
When you put up the botany jacket's hood, this happens 
http://puu.sh/kMNa5/26a8bc207b.png

### PR DESCRIPTION
**Pull request by phil235**
- Fixes winter coat hood sprite appearing as a bucket. Fixes #12448
- <b>Removing Smile, Swedish, Chav and Elvis from genetics (taken from #12041). These mutations can still be acquired via adminspawned dna injector. Added a dna injector for laser eyes mutation.</b>
- Fixes using razor on human species without hair shaving non existent hair. Fixes #12393
- Fixes chair deconstruction dropping too much metal. Fixes #12437
- Adding some necessary check for mob buckling in gibber, processor and monkey recycler code, and in ventcrawling proc to fix being able to enter pipe while feeding as a slime. Fixes #12410
- Fixes snapcorn not giving seeds. Fixes #12423
- Fixes portable chem dispenser. Fixes #12444

### CODE REVIEW

**Comment by Razharas** [comment]
On one hand i never expected there to exist mutations that dont have their respective dna block but being obtainable and id consider such a usage of code an abuse of unintended side effect of mechanics
On the other hand im glad datum mutations are so good you can do such things with them

File: code/game/objects/items/weapons/cosmetics.dm [hunk 1]
```diff
@@ -120,7 +120,9 @@
 /obj/item/weapon/razor/attack(mob/M, mob/user)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-
+		if(H.dna.species.id != "human")
+			user << "<span class='warning'>There is nothing to shave!</span>"
+			return
```
Comment (Incoming5643) [inline code/game/objects/items/weapons/cosmetics.dm]: Use

if(!(HAIR in H.dna.species.specflags) || !(FACEHAIR in H.dna.species.specflags)) //Might be a better way to phrase that

There is a mutant race that has hair (slimepeople) and it's just more futureproof in general

Comment (phil235) [inline code/game/objects/items/weapons/cosmetics.dm]: will do

**Comment by PKPenguin321** [comment]
>Added a dna injector for laser eyes mutation.

time to trade those TCs

**Comment by newfren** [comment]
Praise Phil

**Comment by AnonymousNow** [comment]
I don't this isn't the place to request features, buHEALING HANDS WHEN

**Comment by TheDreamweaver** [comment]
Does your winter coat fix work for all hooded items? For instance the priest hoody was giving bucket head too.

**Comment by phil235** [comment]
@TheDreamweaver It should fix it, yes.

**Comment by TheDreamweaver** [comment]
@AnonymousNow Would you want it as a beneficial genetic trait? And could you use it on yourself?

**Comment by AnonymousNow** [comment]
Yes, and not necessarily.

**Comment by KorPhaeron** [comment]
@phil235 

Genetics no longer functions

**Comment by TheDreamweaver** [comment]
@KorPhaeron Wait what, what do you mean it doesn't function?

**Pull request merged**